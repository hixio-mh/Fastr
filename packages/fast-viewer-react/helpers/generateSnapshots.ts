/* tslint:disable:no-console */
import React = require("react");
import renderer = require("react-test-renderer");

export interface ExampleProps {
    component: React.ReactNode;
    data: any;
    styles: string;
}

export interface Example {
    title: string;
    props?: ExampleProps;
    children?: React.ReactNode;
}

export interface Examples {
    data: Example[];
}

/**
 * Automates taking snapshot tests from code examples.
 * @param {Array} examples - a set of example objects that contain a title, a
 * @param {Function} reactComponent - The react component to render
 * react component, props for that component, and children for that component
 */
export default (examples: Examples, reactComponent: React.ComponentClass<any>): void => {
    if (!reactComponent) {
        console.error("No component param passed to generateSnapshots.");
        return;
    }

    if (!examples) {
        console.error("No examples param passed to generateSnapshots.");
        return;
    }

    if (!examples.data.length) {
        console.warn(
            "No examples found - ensure your example exports an object with a `data` member containing an array of component examples."
        );
    }

    for (const example of examples.data) {
        test(
            example.title,
            (): void => {
                const component: any = renderer.create(
                    React.createElement(reactComponent, example.props, example.children)
                );

                const json: any = component.toJSON();

                expect(json).toMatchSnapshot();
            }
        );
    }
};
