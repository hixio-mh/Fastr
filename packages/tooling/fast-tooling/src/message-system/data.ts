import { uniqueId } from "lodash-es";
import {
    Data,
    DataDictionary,
    LinkedData,
    LinkedDataDictionaryConfig,
    LinkedDataDictionaryUpdate,
} from "./data.props";

/**
 * Resolves a set of data dictionaries from multiple data items
 */
function resolveDataDictionaries(
    parentId: string,
    parentDataLocation: string,
    dataSet: Data<unknown>[]
): DataDictionary<unknown>[] {
    return dataSet.map(dataItem => {
        /* eslint-disable-next-line @typescript-eslint/no-use-before-define */
        return resolveDataDictionary(parentId, parentDataLocation, dataItem, {});
    });
}

/**
 * Resolves a data dictionary from a data item
 */
function resolveDataDictionary(
    parentId: string,
    parentDataLocation: string,
    data: Data<unknown>,
    itemDictionary: { [key: string]: Data<unknown> }
): DataDictionary<unknown> {
    const id: string = uniqueId("fast");
    const dataDictionary: DataDictionary<unknown> = [itemDictionary, id];
    const linkedDataDictionary: DataDictionary<unknown>[] | null = Array.isArray(
        data.linkedData
    )
        ? resolveDataDictionaries(id, parentDataLocation, data.linkedData)
        : null;

    const currentData = data.data;

    if (linkedDataDictionary !== null) {
        linkedDataDictionary.forEach(linkedDataDictionaryItem => {
            const currentLinkedDataItems: LinkedData[] =
                (currentData as any)[
                    linkedDataDictionaryItem[0][linkedDataDictionaryItem[1]].parent
                        .dataLocation
                ] || [];

            (currentData as any)[
                linkedDataDictionaryItem[0][
                    linkedDataDictionaryItem[1]
                ].parent.dataLocation
            ] = currentLinkedDataItems.concat([
                {
                    id: linkedDataDictionaryItem[1],
                },
            ]);
            dataDictionary[0] = {
                ...dataDictionary[0],
                ...linkedDataDictionaryItem[0],
            };
        });
    }

    dataDictionary[0][id] = {
        schemaId: data.schemaId,
        data: currentData,
        parent: {
            id: parentId,
            dataLocation: data.dataLocation || parentDataLocation,
        },
    };

    return dataDictionary;
}

export function getLinkedDataDictionary(
    config: LinkedDataDictionaryConfig
): LinkedDataDictionaryUpdate {
    const resolvedDataDictionary = resolveDataDictionaries(
        config.dictionaryId,
        config.dataLocation,
        config.linkedData
    );

    return {
        dataDictionary: resolvedDataDictionary[0],
        dictionaryId: config.dictionaryId,
    };
}
