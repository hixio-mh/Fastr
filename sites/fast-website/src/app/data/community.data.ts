import DiscordIcon from "svg/icon-discord.svg";
import GithubIcon from "svg/icon-github.svg";
import TwitterIcon from "svg/icon-twitter.svg";

export interface CommunityContentPlacementData {
    actionLink: string;
    actionText: string;
    body: string;
    header: string;
    icon: string;
}

export const communityContentPlacementData: CommunityContentPlacementData[] = [
    {
        actionLink: "https://discord.gg/FcSNfg4",
        actionText: "Join the Discord Chat",
        body:
            "Join our active community on Discord. Follow the latest updates and contributions, ask questions, give feedback, or keep up on our reading list.",
        header: "Discord",
        icon: DiscordIcon,
    },
    {
        actionLink: "https://twitter.com/fast_dna",
        actionText: "Follow us on Twitter",
        body:
            "Follow along as we share out the latest happenings on Twitter. You will find important updates, announcements, and sneak peeks.",
        header: "Twitter",
        icon: GithubIcon,
    },
    {
        actionLink: "https://github.com/microsoft/fast-dna",
        actionText: "Get Started on Github",
        body:
            "Explore the FAST repository on Github and try out our components, utilities, and tools. Or, mix-and-match with your own solutions.",
        header: "Github",
        icon: TwitterIcon,
    },
];
