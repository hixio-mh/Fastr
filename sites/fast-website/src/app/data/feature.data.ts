export interface FeatureCardData {
    body: string;
    documentationLink: string;
    githubLink: string;
    header: string;
}

export const featureCardData: FeatureCardData[] = [
    {
        body:
            "Create new component compositions by nesting, styling base components, and passing unhandled props. The combinations are endless.",
        documentationLink: "#",
        githubLink: "#",
        header: "Composition",
    },
    {
        body:
            "Use Fluent by default or customize design system properties to make it your own. Or, create your own design system to use with FAST components.",
        documentationLink: "#",
        githubLink: "#",
        header: "Design Systems",
    },
    {
        body:
            "Out of the box, FAST components are built on React, but you can build components on any modern framework using the FAST system.",
        documentationLink: "#",
        githubLink: "#",
        header: "Technology",
    },
    {
        body:
            "All FAST components follow WCAG 2.1, are W3C spec-compliant and use the W3C interaction models when available.",
        documentationLink: "#",
        githubLink: "#",
        header: "Web Standards",
    },
    {
        body:
            "Design sophisticated animation sequences with the animation library, an interface for the Web Animations API.",
        documentationLink: "#",
        githubLink: "#",
        header: "Animation",
    },
    {
        body:
            "Create color palettes, extract colors from images, and handle other color operations using our color library.",
        documentationLink: "#",
        githubLink: "#",
        header: "Colors",
    },
    {
        body:
            "Build layouts such as a 12 column grid for content or an application grid with resizable panels.",
        documentationLink: "#",
        githubLink: "#",
        header: "Layout / Grid",
    },
    {
        body:
            "Leverage a toolkit of general utilities such as keyboarding, Right-To-Left (RTL), number, and string manipulation.",
        documentationLink: "#",
        githubLink: "#",
        header: "Other",
    },
];
