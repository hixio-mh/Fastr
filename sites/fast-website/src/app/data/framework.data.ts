export interface FrameworkContentPlacementData {
    body?: string;
    header?: string;
    headerSubscript?: string;
}

export const frameworkContentPlacementData: FrameworkContentPlacementData[] = [
    {
        body:
            "Angular's default binding syntax will always set properties on an element. This works well for rich data, like objects and arrays, and also works well for primitive values so long as the Custom Element author has mapped any exposed attributes to corresponding properties.",
        header: "Angular",
        headerSubscript: "8.2.14",
    },
    {
        body:
            "DIO uses a runtime heuristic to determine if it should pass data to Custom Elements as either properties or attributes. If a property is already defined on the element instance, DIO will use properties, otherwise it will fallback to attributes.",
        header: "DIO",
    },
    {
        body:
            "hyperHTML will pass data to an element as properties, as long as the property is defined on the element's prototype. Otherwise it will fallback to passing data as attributes.",
        header: "hyperHTML",
        headerSubscript: "2.32.2",
    },
    {
        body:
            "Dojo will pass data as attributes only when the data is a type of string, otherwise it is set as a property.",
        header: "Dojo",
        headerSubscript: "6.0.4",
    },
    {
        body:
            "Hybrids will pass data to an element as properties, as long as the property is defined on the element's prototype. Otherwise it will fallback to passing data as attributes.",
        header: "hybrids",
        headerSubscript: "4.1.7",
    },
    {
        body:
            "AngularJS can declaratively pass data to attributes using ng-attr, or to properties using ng-prop.",
        header: "AngularJS (1.x)",
        headerSubscript: "1.7.9",
    },
];
