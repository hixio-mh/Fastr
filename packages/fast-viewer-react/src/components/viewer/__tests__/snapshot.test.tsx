import Viewer from "../";
import data from "./data";
import generateSnapshots from "../../../../helpers/generateSnapshots";

generateSnapshots(data, Viewer);
