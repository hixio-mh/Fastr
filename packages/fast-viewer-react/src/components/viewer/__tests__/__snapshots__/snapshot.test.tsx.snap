// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
<div>
  <base
    target="_blank"
  />
  <iframe
    className="iframe-0-0-3"
    style={
      Object {
        "height": "0px",
      }
    }
  >
    Your browser does not support iframes.
  </iframe>
</div>
`;
